app/controllers/CacheController.scala

```diff
@@ -46,7 +48,7 @@ class CacheController @Inject() (
   def get(lrn: String): Action[AnyContent] =
     getUserAnswers[UserAnswers](lrn)(identity)
 
-  def post(lrn: String): Action[JsValue] = authenticateAndLock(lrn).async(parse.json) {
+  def post(lrn: String): Action[JsValue] = (authenticateAndLock(lrn) andThen getVersion).async(parse.json) {
```

Comment:
POST requests probably don't need changing. We've set the isTransitional flag on the PUT and we read it as true if it doesn't exist.

What we should be looking to change is the GET. In traditional API design we would return the 4xx response in the API, rather than in the frontend. Think of it as being all the frontends we want to restrict access to a transition document in final mode in one go. We then decide in the frontend how to handle that 4xx response from the API.

Reply:
Ok, ignore my comment about the POST request. I can't actually recall why I mentioned that 😆